universal-login-react/src/ui/TopUp/Fiat/FiatFooter.tsx

```diff
@@ -12,16 +12,18 @@ import {TopUpProvider} from '../../../core/models/TopUpProvider';
 import {getMinimalAmount} from '../../../core/utils/getMinimalAmountForFiatProvider';
 import {useAsync} from '../../hooks/useAsync';
 import {InfoText} from '../../commons/Text/InfoText';
+import {ETHER_NATIVE_TOKEN} from '@unilogin/commons';
 
 interface FiatFooterProps {
   walletService: WalletService;
   paymentMethod?: TopUpProvider;
   selectedCurrency: string;
 }
-
 export const FiatFooter = ({paymentMethod, walletService, selectedCurrency}: FiatFooterProps) => {
+  const currencyAddress = walletService.sdk.tokensDetailsStore.getTokenAddress(selectedCurrency);
+  const currencyDetails = useAsync(() => walletService.sdk.tokenDetailsService.getTokenDetails(currencyAddress || ETHER_NATIVE_TOKEN.address), [])[0];
```

Comment:
We can add function to `tokensDetailsStore` `getTokenDetailsBySymbol`. Btw `getTokenDetails` is not async, so we don't need to use `useAsync` function

Reply:
`async getTokenDetails(tokenAddress: string): Promise<TokenDetails> {
    if (tokenAddress === this.saiTokenAddress) {
      return {address: this.saiTokenAddress, symbol: 'SAI', name: 'Sai Stablecoin v1.0'};
    }
    const symbol = await this.getSymbol(tokenAddress);
    const name = await this.getName(tokenAddress);
    return {address: tokenAddress, symbol, name};
  }`